File: backend/scripts/create_admin.js
require('dotenv').config({ path: './.env' });
const adapter = require('../src/db/adapter');
const bcrypt = require('bcryptjs');
const readline = require('readline');

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

function question(q) {
  return new Promise(resolve => rl.question(q, answer => resolve(answer)));
}

async function run() {
  try {
    const username = (process.env.ADMIN_USERNAME) || await question('Admin username [admin]: ');
    const email = (process.env.ADMIN_EMAIL) || await question('Admin email [admin@digitalbazar.com]: ');
    const password = (process.env.ADMIN_PASSWORD) || await question('Admin password: ');

    if (!password) {
      console.error('Password required');
      process.exit(1);
    }

    await adapter.connectMongo();

    const existing = await adapter.getUserByUsername(username) || await adapter.getUserByUsername(email);
    if (existing) {
      console.error('User already exists');
      process.exit(1);
    }

    const id = `user-${Date.now()}`;
    const user = { id, username: username || 'admin', email: email || 'admin@digitalbazar.com', full_name: 'Administrator', is_admin: true, created_at: new Date().toISOString() };
    const hash = bcrypt.hashSync(password, 10);
    await adapter.createUser(user, hash);
    console.log('Created admin user', user.username);
    process.exit(0);
  } catch (e) {
    console.error('ERROR', e);
    process.exit(1);
  }
}

run();

File: backend/create_admin.py
import asyncio
import os
import uuid
from datetime import datetime, timezone
from motor.motor_asyncio import AsyncIOMotorClient
from passlib.context import CryptContext
from dotenv import load_dotenv
from pathlib import Path

# Load environment variables
ROOT_DIR = Path(__file__).parent
load_dotenv(ROOT_DIR / '.env')

# MongoDB connection
mongo_url = os.environ.get('MONGO_URL')
db_name = os.environ.get('DB_NAME', 'digital_bazar_test')

if not mongo_url:
    print("Error: MONGO_URL not found in environment variables.")
    exit(1)

client = AsyncIOMotorClient(mongo_url)
db = client[db_name]

# Password hashing
pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")

def hash_password(password: str) -> str:
    return pwd_context.hash(password)

async def create_admin():
    username = input("Enter admin username [admin]: ") or "admin"
    email = input("Enter admin email [admin@digitalbazar.com]: ") or "admin@digitalbazar.com"
    password = input("Enter admin password: ")
    full_name = input("Enter admin full name [Administrator]: ") or "Administrator"

    if not password:
        print("Error: Password is required.")
        return

    # Check if user already exists
    existing_user = await db.users.find_one({"$or": [{"username": username}, {"email": email}]})
    if existing_user:
        print(f"Error: User with username '{username}' or email '{email}' already exists.")
        return

    user_id = str(uuid.uuid4())
    hashed_password = hash_password(password)

    user_doc = {
        "id": user_id,
        "username": username,
        "email": email,
        "full_name": full_name,
        "is_admin": True,
        "created_at": datetime.now(timezone.utc)
    }

    # Save user and password
    await db.users.insert_one(user_doc)
    await db.user_passwords.insert_one({"user_id": user_id, "password_hash": hashed_password})

    print(f"Successfully created admin user: {username} ({email})")

if __name__ == "__main__":
    asyncio.run(create_admin())
